### ISSUE TITLE
Make the charged energy (`wh`) sensor energy dashboard compatible

### ISSUE THREAD

**Issue by KlausPuchner** [syssi/homeassistant-goecharger-mqtt#50]
The sensor in the go-e integration is designed to give wh:
![image](https://user-images.githubusercontent.com/43047262/183156396-18d0f59a-627a-40c3-8c96-c07c4bb40055.png)

Unfortunately this does not seem to work well in the home assistant energy dashboard since it seems to expect kwh:
![image](https://user-images.githubusercontent.com/43047262/183156575-57cee9ed-1087-4f46-b2c9-0fb218516b5f.png)

Is it somehow possible to make home assistant read the values correctly so that it is also correctly shown?

**Comment by KlausPuchner** [comment]
I just found that in the energy dashboard setting there is a warning/error hidden:

![image](https://user-images.githubusercontent.com/43047262/183157621-90f0316e-1f97-4438-8da3-d323628d0986.png)

No idea if this has something to do with my issue mentioned above

**Comment by syssi** [comment]
Which Home Assistant version do you use? I will try to reproduce the issue with a fresh Home Assistant instance (2022.8.0) tomorrow.

**Comment by KlausPuchner** [comment]
Home Assistant 2022.7.5
Frontend 20220707.1 - latest

**Comment by syssi** [comment]
Which version of this custom component do you use?

**Comment by syssi** [comment]
I made some progress: It looks like you are the first one using the key `wh` at the energy dashboard. More common is using `eto` (total energy). Could you do me a favor? Please check the history of your `wh` sensor. Does the value reset once a day or does it increase without resets? Could you try to explain the difference between `eto`  (total energy)  and `wh` (energy charged)?

Is this assumption correct: If you charge your car 3 times a day the `wh` sensor increases 3 times from `0` to `charged energy`?

**Comment by syssi** [comment]
I've pushed a fix which should improve the situation (hopefully): https://github.com/syssi/homeassistant-goecharger-mqtt/commit/9274c8509e5ff5b30e6133e11867069490d18da8

**Comment by syssi** [comment]
I've published some fake traffic using

```
I=0
while true;
do
  I=$(($I+1000))
  mosquitto_pub -r -t /go-eCharger/111111/wh -m $I
  sleep 60
  echo $I
done
```

and checked the energy dashboard 1-2 hours later:

![a](https://user-images.githubusercontent.com/2735933/183244447-ca1de6c6-c576-4c37-b67c-e6dd3e6b395a.png)
![b](https://user-images.githubusercontent.com/2735933/183244445-ca8ab36c-2b49-41a4-9d1d-70a10b71ee44.png)
![c](https://user-images.githubusercontent.com/2735933/183244446-e467a8ee-41c5-47ec-96bd-8e3f60fd689f.png)

On my side it looks fine now. I will prepare a release now. Please update your custom component to the new version and close the issue if it's solved.

**Comment by KlausPuchner** [comment]
Hi! Thank you for your incredible support! :) And sorry for my late response :(

I just updated the custom component to 1.12.1 and home assistant to 2022.8.1 (which was requested due to compatibility issues).

Unfortunately the energy dashboard still does not show the charged energy amount (it seems to still get it in wh instead of kwh), although the weird warning/error message I mentioned earlier did disappear now.

Regarding eto (total energy) vs. charged energy: I can not choose any sensor named eto in the energy dashboard configuration. I also cannot tell the difference beteeen eto and charged energy, but imo only charged energy makes sense in the case of the energy dashboard (since it is only necessary to include the actually consumed power).

Loading multiple times should work, but unfortunately it seems I have lost the old charging data when I readded the charged energy sensor to the energy board.
